src/control/tag.rs

```diff
@@ -3,50 +3,79 @@ use core::{fmt, mem};
 /// Single tag in a control group.
 #[derive(Copy, Clone, PartialEq, Eq)]
 #[repr(transparent)]
-pub(crate) struct Tag(pub(super) u8);
+pub(crate) struct Tag(pub(super) i8);
 impl Tag {
     /// Control tag value for an empty bucket.
-    pub(crate) const EMPTY: Tag = Tag(0b1111_1111);
+    pub(crate) const EMPTY: Tag = Tag(0b0111_1111); // 127
+    pub(crate) const EMPTY32: i32 = 0x7F7F7F7F;
```

Comment:
There should probably be a separate `ExpandedTag` type for this rather than a plain `i32`.